code/modules/antagonists/zombie/zombie.dm

```diff
@@ -0,0 +1,429 @@
+
+
+
+
+/datum/action/innate/zombie/talk
+	name = "Chat"
+	desc = "Chat with your fellow infected."
+	icon_icon = 'icons/mob/actions/actions_cult.dmi'
+	button_icon_state = "cult_comms"
+
+/datum/action/innate/zombie/talk/Activate()
+	var/input = stripped_input(usr, "Please choose a message to tell to the other zombies.", "Infected Communications", "")
+	if(!input || !IsAvailable())
+		return
+
+	talk(usr, input)
+
+/datum/action/innate/zombie/talk/proc/talk(mob/living/user, message)
+	var/my_message
+	if(!message)
+		return
+	var/title = "Zombie"
+	var/span = "cult"
+	my_message = "<span class='[span]'><b>\[[title]] [findtextEx(user.name, user.real_name) ? user.name : "[user.real_name] (as [user.name])"]:</b> [message]</span>"
+	for(var/i in GLOB.player_list)
+		var/mob/M = i
+		if(isinfected(M))
+			to_chat(M, my_message)
+		else if(M in GLOB.dead_mob_list)
+			var/link = FOLLOW_LINK(M, user)
+			to_chat(M, "[link] [my_message]")
+
+	user.log_talk(message, LOG_SAY, tag="zombie")
+
+/datum/action/innate/zombie/choose_class
+	name = "Evolve"
+	desc = "Evolve into a special class."
+	icon_icon = 'icons/mob/actions/actions_cult.dmi'
+	button_icon_state = "cultfist"
+
+/datum/action/innate/zombie/choose_class/Activate()
+	var/selected = input(usr, "Choose a class to evolve into", "Evolution") as null|anything in list("Runner", "Juggernaut", "Spitter")
+	if(!selected || !IsAvailable())
+		return
+	if(!isinfectedzombie(owner))
+		return
+	evolve(selected)
+
+/datum/action/innate/zombie/choose_class/IsAvailable(forced = FALSE)
+	if(!isinfected(owner))
+		return
+	var/datum/antagonist/zombie/Z = locate() in owner.mind.antag_datums
+
+	if(Z.class_chosen && !forced)
+		return FALSE
+	return ..()
+
+/datum/action/innate/zombie/choose_class/proc/evolve(class)
+	var/mob/living/carbon/human/H = owner
+	var/datum/antagonist/zombie/Z = locate() in owner.mind.antag_datums
+	if(!isinfected(H))
+		return
+	switch(class)
+		if("Runner")
+			H.set_species(/datum/species/zombie/infectious/gamemode/runner)
+			Z.adren = new()
+			H.AddAbility(Z.adren)
+			to_chat(owner, "<span class='warning'>You can now run, and your movement speed is considerably faster. You do less damage and can take less damage though.</span>")
+		if("Juggernaut")
+			H.set_species(/datum/species/zombie/infectious/gamemode/juggernaut)
+			Z.tank = new()
+			H.AddAbility(Z.tank)
+			to_chat(owner, "<span class='warning'>You can now take quite a beating, and heal a bit slower.</span>")
+		if("Spitter")
+			H.set_species(/datum/species/zombie/infectious/gamemode/spitter)
+			Z.spit = new()
+			Z.acid = new()
+			H.AddAbility(Z.spit)
+			H.AddAbility(Z.acid)
+			to_chat(owner, "<span class='warning'>You can now right click on walls and doors, and cover them in acid! You are weaker in combat though.</span>")
+
+	owner.visible_message("<span class='danger'>[owner] suddenly convulses, as [owner.p_they()] evolve into a [class]!</span>", "<span class='alien'>You have evolved into a [class]</span>")
```

Comment:
Lower case the `class` for proper text capitalization

Reply:
what

Reply:
Juggernaut and whatever is capitalized, itd going to look weird in game

Reply:
Actually intended, used to emphasize. I'd like to keep it capitalized if possible